src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java

```diff
@@ -122,9 +123,31 @@ private static void handlePlanTripResponse(Request request, OtpDispatcherRespons
         long tripStorageStartTime = DateTimeUtils.currentTimeMillis();
 
         Auth0Connection.checkUser(request);
-        Auth0UserProfile profile = Auth0Connection.getUserFromRequest(request);
+        RequestingUser requestingUser = Auth0Connection.getUserFromRequest(request);
+        if (requestingUser == null) {
+            return;
+        }
+
+        // Determine if the Otp request is being made by an actual Otp user or by a third party on behalf of an Otp user.
+        OtpUser otpUser;
+        if (requestingUser.isThirdPartyUser()) {
+            // Api user making a trip request on behalf of an Otp user. In this case, the Otp user id must be provided
+            // as a query parameter.
+            String userId = request.queryParams("userId");
```

Comment:
We probably need a swagger documentation for this field.

Reply:
Yep!

Reply:
Have updated with comment for review.